policy/modules/system/systemd.fc

```diff
@@ -51,6 +51,7 @@ HOME_DIR/\.config/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_unit
 /usr/lib/systemd/system/.*shutdown.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/.*suspend.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/systemd-userdbd\.(service|socket)		--	gen_context(system_u:object_r:systemd_userdbd_unit_file_t,s0)
+/usr/lib/systemd/system-sleep(/.*)?		gen_context(system_u:object_r:systemd_sleep_exec_t,s0)
```

Comment:
I'm not convinced that we want to label also the directory with the `systemd_sleep_exec_t` type... At the very least, `systemd_sleep_t` will now need new permissions to read the directory.

Also, won't this cause a regression for the tlp script? Depending on what it does exactly, systemd_sleep_t might not be a good label for it.

Ideally, each service that installs some script there would have its own "sleep script" domain and file label and would register a transition from `systemd_sleep_t` to this domain. And for custom user scripts there could be a special customizable type (see `customizable_types(5)`) with a transition to some near-unconfined domain (or just `unconfined_service_t`) so they can get their scripts to work easily. Otherwise we'll be playing an infinite catch-up game with what users put there and will want to work and will end up with bloated policy for `systemd_sleep_t`...

Reply:
I changed the PR especially for `/usr/lib/systemd/system-sleep` is expected to be a place for user scripts.